### PR TITLE
Wms wmts

### PR DESCRIPTION
**Pull request by samgehret**
Created configurable WMS Raster Tile URL. Although could only test under limited circumstances, using the following URL:

https://geodata.state.nj.us/imagerywms/Natural2015?bbox={bbox-epsg-3857}&format=image/png&service=WMS&version=1.1.1&request=GetMap&srs=EPSG:3857&width=256&height=256&layers=Natural2015

Parameters of the raster tileset must be specified as parameters in the input URL (as opposed to standalone setting entries in the GUI.

### CODE REVIEW

File: pbiviz.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "visual": {
     "name": "mapboxGLMap",
     "displayName": "Mapbox Visual",
-    "guid": "PBI_CV_EB3A4088_75C5_4746_9D8B_255A7B7ECD6C",
+    "guid": "PBI_CV_EB3A4088_75C5_4746_9D8B_255A7B7ECD6C_6",
```
Comment (ryanbaumann) [inline pbiviz.json]: We'll want to revert this to the orig guid value for merge to master.

File: src/ts/datasources/raster.ts [hunk 2]
```diff
@@ -6,26 +6,22 @@ module powerbi.extensibility.visual.data {
...
         constructor() {
-            super()
-            console.log('testing datasource')
+            super('wms')
         }
 
         addSources(map, settings) {
-            console.log('testing add source for raster')
-            console.log(settings)
+            console.log(settings.raster.tilesize)
```
Comment (ryanbaumann) [inline src/ts/datasources/raster.ts]: remove console log info statements before merge

File: src/ts/layers/raster.ts [hunk 3]
```diff
@@ -8,24 +8,20 @@ module powerbi.extensibility.visual {
...
             const map = this.parent.getMap();
             const rasterLayer = mapboxUtils.decorateLayer({
                 id: 'raster',
                 source: 'raster',
                 type: 'raster'
             });
-            console.log('before layer id')
-            console.log(beforeLayerId)
-            map.addLayer(rasterLayer, 'waterway-label');
+            map.addLayer(rasterLayer, 'water');
```
Comment (ryanbaumann) [inline src/ts/layers/raster.ts]: We should update the `water` layer here to be `beforeLayerId`.

**Comment by samgehret** [comment]
@ryanbaumann made changes addressing your review

File: src/ts/layers/raster.ts [hunk 4]
```diff
@@ -21,7 +21,7 @@ module powerbi.extensibility.visual {
                 source: 'raster',
                 type: 'raster'
             });
-            map.addLayer(rasterLayer, 'water');
+            map.addLayer(rasterLayer, beforeLayerId);
```
Comment (ryanbaumann) [inline src/ts/layers/raster.ts]: @samgehret testing this out, where do dataviz layers get placed in relation to the WMS/WMTS layer?  Above this raster map layer?

File: src/ts/datasources/raster.ts [hunk 5]
```diff
@@ -10,14 +10,13 @@ module powerbi.extensibility.visual.data {
         }
 
         addSources(map, settings) {
-            console.log(settings.raster.tilesize)
             map.addSource('raster', {
                 type: 'raster',
                 tiles: [
                     settings.raster.url
                 ]
                 // tileSize: 256
```
Comment (ryanbaumann) [inline src/ts/datasources/raster.ts]: @samgehret does tileSize need to be configurable by the end user?  Most WMS sources will generally be 256px outside of Mapbox, but the user may need to configure it.

Comment (samgehret) [reply src/ts/datasources/raster.ts]: @ryanbaumann Currently, this is handled as a parameter in the URL. Like this https://geodata.state.nj.us/imagerywms/Natural2015?bbox={bbox-epsg-3857}&format=image/png&service=WMS&version=1.1.1&request=GetMap&srs=EPSG:3857&width=256&height=256&layers=Natural2015

Comment (ryanbaumann) [reply src/ts/datasources/raster.ts]: @samgehret the mapboxgl source component could expose a configurable as well.  Then developers will be able to add whichever tile size their WMS / WMTS source supports.

**Comment by RoryCT** [comment]
Not sure what the etiquette is here, sorry - Mapbox support suggested I comment here with my use case. We're looking to move to Power BI with Mapbox for geospatial reporting, but require our own WMS server layers to be available. Mapbox's custom visual capabilities outstrip all of the other offerings, but this feature is what's holding up change (currently sticking with an ageing and monolithic but fit-for-purpose solution). Is this feature likely to be incorporated soon? Can I help at all?

**Comment by ryanbaumann** [comment]
Thanks for chiming in @RoryCT!  We're scoping the next release priorities right now, and will consider your feature request needs as part of our scoping.  cc/ @samgehret

**Comment by ryanbaumann** [comment]
Close in favor of https://github.com/mapbox/mapboxgl-powerbi/pull/288

**Pull request closed without merging**